map_model/src/objects/lane.rs

```diff
@@ -405,13 +409,20 @@ impl LaneSpec {
             // https://en.wikipedia.org/wiki/Lane#Lane_width
             LaneType::Driving => {
                 let mut choices = vec![
-                    (Distance::feet(8.2), "narrow"),
+                    (Distance::feet(8.0), "narrow"),
                     (SERVICE_ROAD_LANE_THICKNESS, "alley"),
                     (Distance::feet(10.0), "typical"),
-                    (Distance::feet(12.0), "highways"),
+                    (Distance::feet(12.0), "highway"),
                 ];
-                if rank == osm::RoadRank::Highway {
+                if rank == osm::RoadRank::Highway
+                    && tags
+                        .get(osm::HIGHWAY)
+                        .map(|x| !x.ends_with("_link"))
+                        .unwrap_or(true)
+                {
                     choices.rotate_right(1);
+                } else if tags.is(osm::HIGHWAY, "service") || tags.is("narrow", "yes") {
```

Comment:
What is the goal of juggling the choices around?

A couple things that seemed strange to me:


entries: 8, 5, 10, 12, but I think I expect [5, 8, 10, 12], so that they'd be  monotonically increasing:
<img width="324" alt="Screen Shot 2021-05-04 at 1 27 09 PM" src="https://user-images.githubusercontent.com/217057/117065670-a373f400-acdc-11eb-93e7-8ef90844a7cb.png">

It seems a little inexplicable that 8ft is in there twice:
<img width="319" alt="Screen Shot 2021-05-04 at 1 26 02 PM" src="https://user-images.githubusercontent.com/217057/117065673-a4a52100-acdc-11eb-96a5-fb1ae4fdc3c5.png">

Reply:
> What is the goal of juggling the choices around?

Should be fixed. From `typical_lane_widths`, the first is used as the default in the map import, so juggling is necessary. But the UI now always sorts.

> It seems a little inexplicable that 8ft is in there twice:

One of them is secretly 8.2021 ft (2.5m), but we're rounding by default. This should actually go away after regenerating this particular map; the default width of the selected lane should change slightly.